src/api.rs

```diff
@@ -68,31 +58,43 @@ pub struct UploadFormSingleFile {
 }
 
 #[derive(Clone)]
-pub struct Api {
-    pub ods_client: Arc<OmnectDeviceServiceClient>,
```

Comment:
ServiceClient is non `Arc` anymore. Thus we might have multiple instances of `ServiceClient` and SocketClient as well. As a result the internal caching feature of `hyper_util::client::legacy::Client` isn't used anymore.

Reply:
So we can keep as implemented by you, but then we should ensure `create_module_certificate` gets the IP address (passed) and doesn't create another instance of ods `ServiceClient`